### ISSUE TITLE
Relying on Homebrew configuration paths instead of `/opt/boxen/config`

### ISSUE THREAD

**Issue by jacobbednarz** [boxen/boxen#214]
Whilst trying to update `nginx` in boxen/puppet-nginx#51, I've found that the convention of putting configuration files into `/opt/boxen/config` is causing it to fail on newer versions of OSX (whereas it works fine on older installations).

```
==> ./configure --prefix=/usr/local/Cellar/nginx/1.10.2-boxen1 --with-http_ssl_module --with-pcre --with-ipv6 --with-cc-opt=-I/usr/local/opt/pcre/include -I/usr/local/opt/openssl/include --with-ld-opt=-L/usr/local/opt/pcre/lib -L/usr/local/opt/openssl/lib --conf-path=/opt/boxen/config/nginx/nginx.conf --pid-path=/opt/boxen/data/nginx/nginx.pid --lock-path=/opt/boxen/data/nginx/nginx.lock --with-http_v2_module
==> make
==> make install
Last 15 lines from /Users/dtbaker/Library/Logs/Homebrew/nginx/03.make:

make -f objs/Makefile install
test -d '/usr/local/Cellar/nginx/1.10.2-boxen1' || mkdir -p '/usr/local/Cellar/nginx/1.10.2-boxen1'
test -d '/usr/local/Cellar/nginx/1.10.2-boxen1/sbin' \
		|| mkdir -p '/usr/local/Cellar/nginx/1.10.2-boxen1/sbin'
test ! -f '/usr/local/Cellar/nginx/1.10.2-boxen1/sbin/nginx' \
		|| mv '/usr/local/Cellar/nginx/1.10.2-boxen1/sbin/nginx' \
			'/usr/local/Cellar/nginx/1.10.2-boxen1/sbin/nginx.old'
cp objs/nginx '/usr/local/Cellar/nginx/1.10.2-boxen1/sbin/nginx'
test -d '/opt/boxen/config/nginx' \
		|| mkdir -p '/opt/boxen/config/nginx'
cp conf/koi-win '/opt/boxen/config/nginx'
cp: /opt/boxen/config/nginx/koi-win: Operation not permitted
make[1]: *** [install] Error 1
make: *** [install] Error 2
```

Swapping the `--conf` flag to use `/usr/local/etc` (like upstream Homebrew does) allows the installation to work as expected.

I'm wondering if we want to keep the `/opt/boxen/config` convention or whether we can move the Boxen puppet modules to instead manage the file under `/usr/local/etc`.

@MikeMcQuaid do you have any context on whether this move would cause issues in Homebrew?

**Comment by dgoodlad** [comment]
From a historical perspective, we always wanted to keep boxen-managed things in `/opt/boxen` to improve discoverability and to aid in nuking. However, I think that line's been blurred for a long time now, and this proposal doesn't sound unreasonable.

There are a couple of non-trivial things to consider, though:

* `script/nuke` would now need to know to clear out _some_ configs from `/usr/local/etc`
* inconsistency in where configs are stored will hurt; you'll want to go _all in_, and update all core modules at once.

**Comment by jacobbednarz** [comment]
> we always wanted to keep boxen-managed things in `/opt/boxen` to improve discoverability

This was my primary concern. If we started allowing these configuration files in shared locations it's going to be very difficult to track which system is managing the file and I'm thinking it will introduce race conditions where (for example) boxen runs, updates homebrew which in turn updates the config file and we would be left with the homebrew version of the configuration, not Boxen. We could mitigate this in a couple of ways but it seems pretty hacky to do.

**Comment by MikeMcQuaid** [comment]
This breaks currently because as of Homebrew 1.3.0 you cannot install into arbitrary locations on the disk from inside a formula. As a result an `etc` directory is your only option. I'd suggest `etc/"boxen"` to avoid clashes.

**Comment by jacobbednarz** [comment]
That seems pretty reasonable @MikeMcQuaid, I'll POC it out. Thanks! 🌟

**Comment by jacobbednarz** [comment]
Closing this since it will be addressed in boxen/our-boxen#855
